src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -110,77 +78,78 @@ public class TxPool : ITxPool, IDisposable
-            ThisNodeInfo.AddInfo("Mem est tx   :",
-                $"{(LruCache<Keccak, object>.CalculateMemorySize(32, MemoryAllowance.TxHashCacheSize) + LruCache<Keccak, Transaction>.CalculateMemorySize(4096, MemoryAllowance.MemPoolSize)) / 1000 / 1000}MB"
-                    .PadLeft(8));
+            AddNodeInfoEntryForTxPool();
 
             _transactions = new TxDistinctSortedPool(MemoryAllowance.MemPoolSize, comparer, logManager);
-            _persistentBroadcastTransactions = new TxDistinctSortedPool(MemoryAllowance.MemPoolSize, comparer, logManager);
-            _chainHeadInfoProvider.HeadChanged += OnHeadChange;
-            _ownTimer = new Timer(500);
-            _ownTimer.Elapsed += OwnTimerOnElapsed;
-            _ownTimer.AutoReset = false;
-            _ownTimer.Start();
+            _broadcaster = new TxBroadcaster(comparer, TimerFactory.Default, txPoolConfig, logManager);
+
+            _headInfo.HeadChanged += OnHeadChange;
+
+            _filterPipeline.Add(new NullHashTxFilter());
+            _filterPipeline.Add(new HashCacheBasedTxFilter(_hashCache));
+            _filterPipeline.Add(new MalformedTxFilter(_specProvider, validator, _logger));
+            _filterPipeline.Add(new GasLimitTxFilter(_headInfo, txPoolConfig, _logger));
+            _filterPipeline.Add(new UnknownSenderFilter(ecdsa, _logger));
+            _filterPipeline.Add(new LowNonceFilter(_accounts, _logger));
+            _filterPipeline.Add(new TooFarNonceFilter(txPoolConfig, _accounts, _transactions, _logger));
+            _filterPipeline.Add(new TooExpensiveTxFilter(_headInfo, _accounts, _logger));
+            _filterPipeline.Add(new FeeToLowFilter(_headInfo, _accounts, _transactions, _logger));
+            _filterPipeline.Add(new ReusedOwnNonceTxFilter(_accounts, _nonces, _logger));
+            _filterPipeline.Add(incomingTxFilter ?? NullIncomingTxFilter.Instance);
         }
 
-        private uint FutureNonceRetention  => _txPoolConfig.FutureNonceRetention;
-        
-        internal long? BlockGasLimit { get; set; } = null;
-
-        private UInt256 CurrentBaseFee { get; set; } = 0;
-
         public Transaction[] GetPendingTransactions() => _transactions.GetSnapshot();
-        
+
         public int GetPendingTransactionsCount() => _transactions.Count;
 
         public IDictionary<Address, Transaction[]> GetPendingTransactionsBySender() =>
             _transactions.GetBucketSnapshot();
 
-        public Transaction[] GetOwnPendingTransactions() => _persistentBroadcastTransactions.GetSnapshot();
+        internal Transaction[] GetOwnPendingTransactions() => _broadcaster.GetSnapshot();
 
         private void OnHeadChange(object? sender, BlockReplacementEventArgs e)
         {
+            _hashCache.ClearCurrentBlockCache();
             // we don't want this to be on main processing thread
+            // TODO: I think this is dangerous if many blocks are processed one after another
```

Comment:
@marcindsobczak Yes I think this should be Queued(!), it could then be even processed as batched